### PR TITLE
Feature/composter ai

### LINKED ISSUE #13: Composter [$5]
A new worker who will take biologic material and converts it into compost. Compost can be used like bonemeal (but with less effect) and can be used by the farmer for example.

We need: A barrel (custom block), compost (custom item) and the composter (citizen, building and ai)

[Bountysource](https://www.bountysource.com/issues/36918632-composter?utm_source=36918632&utm_medium=shield&utm_campaign=ISSUE_BADGE)

<bountysource-plugin>

---
There is a **[$5 open bounty](https://www.bountysource.com/issues/36918632-composter?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F43175463&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Fireruner**
Closes #13 

# Changes proposed in this pull request:
- Added worker to use the barrel block
- The worker takes biological material and turns it into compost using the barrel blocks on his building
- The items to be used by the worker can be specified through his building's GUI

Review please

### CODE REVIEW

**Review by LDTTeam-Buildsystem — DISMISSED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 39036 | 19  | 81  | -62  |
| Errors |  592  |  0  |  0  |  0   |
##### More Information:
- [Analysis - Full](https://teamcity.minecolonies.com/viewLog.html?buildId=14115&tab=Inspection&buildTypeId=bt88)

File: src/main/java/com/minecolonies/coremod/tileentities/TileEntityBarrel.java [hunk 1]
```diff
@@ -176,7 +178,7 @@ private boolean checkCorrectItem(final ItemStack itemStack)
                 }
             }
         }
-        return false;
+        return false;**/
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/tileentities/TileEntityBarrel.java]: I think you can remove the commented stuff and only return the above

File: src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java [hunk 2]
```diff
@@ -101,6 +146,12 @@ public ItemStack getSaplingForLeave(final IBlockState block)
         return new ArrayList<>(saplings);
     }
 
+    @Override
+    public  List<ItemStorage> getCopyOfCompostableItems()
+    {
+        return new ArrayList<>(compostableItems);
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/compatibility/CompatibilityManager.java]: Make this an immutable list

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/2701/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingComposter.java [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+    }
+
+    /**
+     * Return a list of barrels assigned to this hut.
+     *
+     * @return copy of the list
+     */
+    public List<BlockPos> getBarrels()
+    {
+        return new ArrayList<>(barrels);
```
Comment (marchermans) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingComposter.java]: Again use an ImmutableList

**Review by marchermans — CHANGES_REQUESTED** [review]
Some small things.

**Review by marchermans — APPROVED** [review]

**Review by LDTTeam-Buildsystem — APPROVED** [review]
#### Analysis Complete
##### Statistics:
|        | Count | New | Old | Diff |
| ------ |:-----:|:---:|:---:|:----:|
| Total  | 39001 |  0  |  6  |  -6  |
| Errors |  592  |  0  |  0  |  0   |
##### More Information:

**Pull request merged**